install.sh

```diff
@@ -8,28 +8,30 @@
 # This script will create links for the dotfiles.
 
 # Step 1: ensure all necessary directories exist before linking files.
-for dir in $(find dotfiles/ -mindepth 1 -type d); do
+for dir in $(ls | find * -mindepth 1 -type d ! -name '.git' ! -name '.gitignore'); do
+
 	target="$HOME/${dir#dotfiles/}"
 	echo "INFO: Ensuring directory exists: $target"
 	mkdir -p $target
-done
 
-# Step 2: create links for all dotfiles
-for file in $(find dotfiles/ -mindepth 1 -type f); do
-	source=$PWD/$file
-	target="$HOME/${file#dotfiles/}"
-	# This if statement could be replaced by passing -f to ln,
-	# but doing so could accidentally delete a dotfile that hasn't been
-	# added to this repo yet, which is why -f is not used, and instead
-	# a warning is printed to inform the user about the situation.
-	if [ -e "$target" ]; then
-		if [ ! -L "$target" ]; then
-			echo "WARN: $target exists but is not a symlink, skipping it."
-		else
-			echo "INFO: $target is already a symlink, skipping it."
-		fi
-	else
-		echo "INFO: linking $source to $target"
-		ln -s $source $target
-	fi
+    # Step 2: create links for all dotfiles
+    for file in $(find dir -mindepth 1 -type f); do
```

Comment:
By nesting this loop, you will be creating the same links multiple times -- you aren't using any variables in the `find` command, so the same commands will be repeated multiple times, which is usually not desirable.

Reply:
Assigning the output of the find to an array will it be the something or to solve this problem we should add a condition check to confirm if the symbolic link exist or not